client/src/Routes.tsx

```diff
@@ -0,0 +1,23 @@
+import { Route, Redirect, Switch } from 'react-router-dom';
+
+import Login from './pages/Login/Login';
+import Signup from './pages/SignUp/SignUp';
+import Dashboard from './pages/Dashboard/Dashboard';
+import ProtectedRoute from './ProtectedRoute';
+
+const Routes = () => {
+  return (
+    <Switch>
+      <Route exact path="/login" component={Login} />
+      <Route exact path="/signup" component={Signup} />
+      <ProtectedRoute exact path="/dashboard">

```

Comment:
i wonder if you can add a property in this HOC called `component` so that it shortens your code and is more similar to Route 

`<ProtectedRoute exact path="/dashboard" component={Dashboard} />`

Reply:
Sure, will do that.